File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if(root == NULL) return {};
        
        queue<TreeNode *> q;
        q.push(root);
        
        while(!q.empty()){
            int l = q.size();
            vector<int> temp;
            for(int i=0; i<l; i++){
                TreeNode *curr = q.front();
                q.pop();
                if(curr){
                    
                 temp.push_back(curr->val);
                q.push(curr->left);
                q.push(curr->right);
                }
            }
            
            if(!temp.empty()) ans.push_back(temp);
            
        }
        
        return ans;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        q = deque([root])
        
        if not root:
            return res
        
        while len(q) > 0:
            n = len(q)
            curr_lvl = []
            
            for _ in range(n):
                node = q.popleft()
                curr_lvl.append(node.val)
                
                for child in [node.left, node.right]:
                    if child is not None:
                        q.append(child)
            
            res.append(curr_lvl)
        
        return res